src/test/java/codeu/model/store/basic/HashtagStoreTest.java

```diff
@@ -0,0 +1,87 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.Hashtag;
+import codeu.model.data.HashtagCreator;
+import codeu.model.data.ModelDataTestHelpers;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.util.HashMap;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class HashtagStoreTest {
+  private HashtagStore hashtagStore;
+  private PersistentStorageAgent mockPersistentStorageAgent;
+
+  @Before
+  public void setup() {
+    mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+    hashtagStore = HashtagStore.getTestInstance(mockPersistentStorageAgent);
+  }
+
+  @Test
+  public void testSetAndGetHashtags() {
+    final HashMap<String, Hashtag> hashtags = new HashMap<String, Hashtag>();
+
+    final Hashtag hash1 =
+        new ModelDataTestHelpers.TestHashtagBuilder().withContent("Soccer").build();
+    final Hashtag hash2 = new ModelDataTestHelpers.TestHashtagBuilder().withContent("Swim").build();
+
+    hashtags.put(hash1.getContent(), hash1);
+    hashtags.put(hash2.getContent(), hash2);
+    hashtagStore.setHashtags(hashtags);
+
+    HashMap<String, Hashtag> resultHashtags = hashtagStore.getAllHashtags();
+    Assert.assertEquals(hashtags, resultHashtags);
+  }
+
+  @Test
+  public void testAddHashtagNotContain() {
+    final HashMap<String, Hashtag> hashtags = new HashMap<String, Hashtag>();
+    final Hashtag hash1 =
+        new ModelDataTestHelpers.TestHashtagBuilder().withContent("Soccer").build();
+    final Hashtag hash2 = new ModelDataTestHelpers.TestHashtagBuilder().withContent("Swim").build();
+
+    UUID user1 = UUID.randomUUID();
+    hash1.addUser(user1);
+    UUID user2 = UUID.randomUUID();
+    hash2.addUser(user2);
+
+    hashtags.put(hash1.getContent(), hash1);
+    hashtags.put(hash2.getContent(), hash2);
+
+    hashtagStore.addHashtag(hash1.getContent(), HashtagCreator.USER, user1);
+    hashtagStore.addHashtag(hash2.getContent(), HashtagCreator.USER, user2);
+
+    HashMap<String, Hashtag> resultHashtags = hashtagStore.getAllHashtags();
+    Assert.assertEquals(
```

Comment:
Instead of using Assert.assertEquals for each fields like this, you should add a function inside ModelDataTestHelpers called `assertHashtagEquals (Hashtag expected, Hashtag actual) {.`